### ISSUE TITLE
rust-analyzer gives an error but cargo check doesn't report an error

### ISSUE THREAD

**Issue by brandon3343** [rust-lang/rust-analyzer#19321]
cargo check and cargo build is working, but rust-analyzer gives an error.

**rust-analyzer version**: 
rust-analyzer version: 0.3.2282-standalone (2df4ecfc74 2025-01-26) 

**rustc version**: 
rustc 1.85.0 (4d91de4e4 2025-02-17)

**editor or extension**: 
vs code: 1.98.0
rust-analyzer extension: 0.3.2330

**repository link **: 
[error_with_diesel](https://github.com/brandon3343/error_with_diesel)

**screenshot**: 

![Image](https://github.com/user-attachments/assets/7b18a7b1-eabd-4734-b3fa-b8c11cd98911)

![Image](https://github.com/user-attachments/assets/67672a00-5e14-4475-af50-b0cca58a7a81)

**Comment by brandon3343** [comment]
Not sure this is a relevant issue or not. https://github.com/rust-lang/rust-analyzer/issues/14607

**Comment by alibektas** [comment]
This is an experimental diagnostics, which means that it has not been thoroughly tested for false positives. I am assuming that you had 

```json
{
    "rust-analyzer.diagnostics.experimental.enable": true,
}
```

in your VSCode settings but forgot to add that to the repro, right?

**Comment by brandon3343** [comment]
Yes, I turned on the setting you mentioned. After turning it off, everything went back to normal. You're amazing, thank you!

**Comment by flodiebold** [comment]
We should still fix this though. We'll have to see after the new trait solver is integrated.

**Comment by Veykril** [comment]
It is implemented here https://github.com/rust-lang/rust-analyzer/blob/5e7dd31c80d5821113ed9c9aa1616a73a63b49a1/crates/hir-ty/src/method_resolution.rs#L874-L921

**Comment by ChayimFriedman2** [comment]
I believe I fixed this case back then.

If you encounter this again please reopen or comment here.
